### PR TITLE
[TACHYON-983] Remove the old lineage code

### PR DESCRIPTION
**Pull request by yupeng9**
https://tachyon.atlassian.net/browse/TACHYON-983

### CODE REVIEW

**Comment by haoyuan** [comment]
Jenkins, test this please.

File: clients/unshaded/src/main/java/tachyon/client/TachyonFS.java [hunk 1]
```diff
@@ -486,7 +485,7 @@ synchronized FileBlockInfo getClientBlockInfo(long blockId) throws IOException {
    * @throws IOException if the underlying master RPC fails
    */
   public synchronized DependencyInfo getClientDependencyInfo(int depId) throws IOException {
```
Comment (gpang) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: should these old lineage methods be deprecated?

File: clients/unshaded/src/main/java/tachyon/client/TachyonFS.java [hunk 2]
```diff
@@ -989,11 +988,7 @@ public synchronized void reportLostFile(long fileId) throws IOException {
    * @throws IOException if the underlying master RPC fails
    */
   public synchronized void requestFilesInDependency(int depId) throws IOException {
```
Comment (gpang) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: should these old lineage methods be deprecated?

File: clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java [hunk 3]
```diff
@@ -212,14 +211,7 @@ public void reportLostFile(TachyonFile file) throws IOException, TachyonExceptio
 
   // TODO: Move this to lineage client
   public void requestFilesInDependency(int depId) throws IOException, TachyonException {
```
Comment (gpang) [inline clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java]: Should this just be removed in favor of the new lineage methods?

File: clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java [hunk 4]
```diff
@@ -392,8 +359,7 @@ public FileStatus getFileStatus(Path path) throws IOException {
   public TachyonFS getTachyonFS() {
     return mTFS;
   }
-
-  @Override
+  
   public URI getUri() {
```
Comment (gpang) [inline clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java]: no more Override?

File: common/src/main/java/tachyon/client/FileSystemMasterClient.java [hunk 5]
```diff
@@ -480,30 +479,6 @@ public synchronized void reportLostFile(long fileId) throws IOException,
   }
 
   /**
-   * Requests files in a dependency.
-   *
-   * @param depId the dependency id
-   * @throws DependencyDoesNotExistException if the dependency does not exist
-   * @throws IOException if an I/O error occurs
-   */
-  public synchronized void requestFilesInDependency(int depId) throws IOException,
```
Comment (gpang) [inline common/src/main/java/tachyon/client/FileSystemMasterClient.java]: Aren't there other lineage-related methods to remove?

File: common/src/thrift/tachyon.thrift [hunk 6]
```diff
@@ -223,21 +212,8 @@ service FileSystemMasterService {
...
-  i32 createDependency(1: list<string> parents, 2: list<string> children,
-      3: string commandPrefix, 4: list<binary> data, 5: string comment, 6: string framework,
-      7: string frameworkVersion, 8: i32 dependencyType, 9: i64 childrenBlockSizeByte)
-    throws (1: InvalidPathException ipe, 2: FileDoesNotExistException fdnee,
-      3: FileAlreadyExistException faee, 4: BlockInfoException bie, 5: TachyonException te)
-
-  DependencyInfo getDependencyInfo(1: i32 dependencyId)
-    throws (1: DependencyDoesNotExistException ddnee)
-
   void reportLostFile(1: i64 fileId)
```
Comment (gpang) [inline common/src/thrift/tachyon.thrift]: Why is this not removed?

**Comment by gpang** [comment]
Overall, looks good. My main question is why some lineage-related methods were not removed. Thanks!

File: clients/unshaded/src/main/java/tachyon/client/TachyonFS.java (hunk 1 above)
Comment (yupeng9) [inline clients/unshaded/src/main/java/tachyon/client/TachyonFS.java]: Sure. But do we need to if `TachyonFS` is deprecated?

File: clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java (hunk 3 above)
Comment (yupeng9) [inline clients/unshaded/src/main/java/tachyon/client/file/TachyonFileSystem.java]: done

File: common/src/main/java/tachyon/client/FileSystemMasterClient.java (hunk 5 above)
Comment (yupeng9) [inline common/src/main/java/tachyon/client/FileSystemMasterClient.java]: done

File: common/src/thrift/tachyon.thrift (hunk 6 above)
Comment (yupeng9) [inline common/src/thrift/tachyon.thrift]: Is used for debugging.

**Comment by yupeng9** [comment]
Jenkins, test this please

File: clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java (hunk 4 above)
Comment (gpang) [inline clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java]: Is this "Override" intentionally removed?

**Comment by gpang** [comment]
I just had a question about a removed "Override", but looks good!

File: clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java (hunk 4 above)
Comment (yupeng9) [inline clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java]: Added it back. Thanks!

**Comment by yupeng9** [comment]
Addressed all the comments. PTAL.

**Comment by gpang** [comment]
LGTM

**Comment by yupeng9** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
@yupeng9 , can you resolve conflicts here and we get rid of the old dependency code on master side?

**Comment by yupeng9** [comment]
Jenkins, test this please

**Comment by calvinjia** [comment]
@yupeng9 Could you resolve the conflicts please?

**Comment by yupeng9** [comment]
@calvinjia , done, PTAL

**Comment by apc999** [comment]
@yupeng9 , could you resolve conflicts for this PR?

**Comment by yupeng9** [comment]
Jenkins, test this please

File: servers/src/main/java/tachyon/master/file/FileSystemMaster.java [hunk 7]
```diff
@@ -56,17 +55,15 @@
 import tachyon.master.file.journal.CompleteFileEntry;
 import tachyon.master.file.journal.DeleteFileEntry;
 import tachyon.master.file.journal.DeleteMountPointEntry;
-import tachyon.master.file.journal.DependencyEntry;
 import tachyon.master.file.journal.InodeDirectoryIdGeneratorEntry;
 import tachyon.master.file.journal.InodeEntry;
 import tachyon.master.file.journal.InodeLastModificationTimeEntry;
 import tachyon.master.file.journal.PersistDirectoryEntry;
 import tachyon.master.file.journal.PersistFileEntry;
+import tachyon.master.file.journal.PersistFileEntry;
```
Comment (apc999) [inline servers/src/main/java/tachyon/master/file/FileSystemMaster.java]: line 63 is identical to line 62

File: servers/src/main/java/tachyon/master/journal/JsonJournalFormatter.java [hunk 8]
```diff
@@ -50,17 +50,16 @@
...
 import tachyon.master.file.journal.DeleteFileEntry;
 import tachyon.master.file.journal.DeleteMountPointEntry;
-import tachyon.master.file.journal.DependencyEntry;
 import tachyon.master.file.journal.InodeDirectoryEntry;
 import tachyon.master.file.journal.InodeDirectoryIdGeneratorEntry;
 import tachyon.master.file.journal.InodeFileEntry;
 import tachyon.master.file.journal.InodeLastModificationTimeEntry;
 import tachyon.master.file.journal.PersistDirectoryEntry;
 import tachyon.master.file.journal.PersistFileEntry;
+import tachyon.master.file.journal.PersistFileEntry;
```
Comment (apc999) [inline servers/src/main/java/tachyon/master/journal/JsonJournalFormatter.java]: ditto

**Comment by apc999** [comment]
+1 LGTM

**Comment by apc999** [comment]
@yupeng9 could you resolve the confliction?

**Comment by yupeng9** [comment]
@apc999 , conflicts resolved, PTAL

**Pull request merged**